GVFS/GVFS.Common/ProductUpgraderInfo.cs

```diff
@@ -104,5 +129,37 @@ private static void RecursiveDelete(string path)
 
             directory.Delete();
         }
+
+        private static Version ReadHighestRecordedAvailableVersion()

```

Comment:
Similar comment here, use `PhysicalFileSystem` and add unit tests.

Reply:
Would bringing in and initialising these dependencies increase the spin-up time for the hooks?

Reply:
Rather than having hooks consume `ProductUpgraderInfo.cs` directly, we could move the method(s) the hooks need to a new `ProductUpgraderInfo.Shared.cs`, and then methods like this one could use `PhysicalFileSystem` without having to pull `PhysicalFileSystem` into the hooks project.